src/main/java/com/codefactorygroup/betting/service/implementations/MarketServiceImpl.java

```diff
@@ -103,4 +103,34 @@ public MarketDTO updateMarket(final MarketDTO newMarket, final Integer marketId)
                 .map(MarketDTO::converter)
                 .orElseThrow(() -> new NoSuchEntityExistsException(String.format("Market with ID=%d doesn't exist.", marketId)));
     }
+
+    @Override
+    public List<Map<String, String>> calcAvgPriceForPreMatchMarkets() {
```

Comment:
ForPrematchMarkets but I can't see any checks to ensure that market is Prematch

Reply:
@mihaidi It was decided later on that for the simplicity the market won't have a property named isPrematch, so there was no way to check it. I just left it like that. I'll change the method name